features/Detaching Unhealthy ASG Instance instead of Terminating

```diff
@@ -0,0 +1,161 @@
+
+Parameters:
+  SNSTopicARN:
+    Type: String
+    Default: ''
+    Description: The ARN of the SNS topic for notifications (optional).
+
+  ASGNames:
+    Type: CommaDelimitedList
+    Description: A comma-separated list of Auto Scaling Group names to monitor (Support multiple ASGs).
+
+Resources:
+  LambdaExecutionRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: lambda.amazonaws.com
+            Action: sts:AssumeRole
+      Policies:
+        - PolicyName: LambdaExecutionPolicy
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - logs:CreateLogGroup
+                  - logs:CreateLogStream
+                  - logs:PutLogEvents
+                  - autoscaling:DescribeAutoScalingGroups
+                  - autoscaling:DetachInstances
+                  - autoscaling:SuspendProcesses
+                  - sns:Publish
+                Resource: '*'
+
+  LambdaFunction:
+    Type: AWS::Lambda::Function
+    Properties:
+      Handler: index.lambda_handler
+      Role: !GetAtt LambdaExecutionRole.Arn
+      Code:
+        ZipFile: |
+          import boto3
+          from datetime import datetime
+          import os
+
+          autoscaling_client = boto3.client('autoscaling')
+          sns_client = boto3.client('sns')
+
+          ASG_NAMES = os.getenv('ASG_NAMES', '').split(',')
+          SNS_TOPIC_ARN = os.getenv('SNS_TOPIC_ARN')
+          MAX_DETACH = 20  # Maximum number of instances to detach at once
+
+          def process_asg(asg_name):
+              response = autoscaling_client.describe_auto_scaling_groups(
+                  AutoScalingGroupNames=[asg_name]
+              )
+              asg = response['AutoScalingGroups'][0]
+              
+              # Check and suspend ReplaceUnhealthy process if not already suspended
+              if 'ReplaceUnhealthy' not in [process['ProcessName'] for process in asg['SuspendedProcesses']]:
+                  autoscaling_client.suspend_processes(
+                      AutoScalingGroupName=asg_name,
+                      ScalingProcesses=['ReplaceUnhealthy']
+                  )
+
+              return [i['InstanceId'] for i in asg['Instances'] if i['HealthStatus'] != 'Healthy'], asg['DesiredCapacity']
+
+          def lambda_handler(event, context):
+              timestamp = datetime.now().strftime('%Y-%m-%d %H:%M:%S')
+              message_details = []
+
+              for asg_name in ASG_NAMES:
+                  unhealthy_instances, desired_capacity = process_asg(asg_name)
+                  max_detach_by_percent = int(desired_capacity * 0.1)
```

Comment:
Please change this line to:
`max_detach_by_percent = max(int(desired_capacity * 0.1),1) # Ensure minimum 1 if there are unhealthy instances. `
Otherwise, when the desired capacity is less than 10. This script won't detach any unhealthy instance. Thank you!